src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs

```diff
@@ -252,13 +252,30 @@ public override void DefaultVisit(Symbol symbol)
                 return;
             }
 
-            bool isPartialMethodDefinitionPart = symbol.IsPartialDefinition(); // CONSIDER: ignore this if isForSingleSymbol?
-            if (isPartialMethodDefinitionPart)
+            bool shouldSkipPartialDefinitionComments = false;
+            if (symbol.IsPartialDefinition())
             {
-                MethodSymbol implementationPart = ((MethodSymbol)symbol).PartialImplementationPart;
-                if ((object)implementationPart != null)
+                if (symbol is not MethodSymbol { PartialImplementationPart: MethodSymbol implementationPart })
```

Comment:
>is not [](http://example.com/codeflow?start=27&length=6)

I think the code will be easier to follow if negation is removed and branches are swapped.

Reply:
Fixed.